### PR TITLE
Add aten::_foreach_copy_

### PR DESCRIPTION
**Pull request by cfgfung**
Add the implementation opeartor _foreach_copy_
Tested on PVC.

### CODE REVIEW

File: src/ATen/native/xpu/ForeachOpList.cpp [hunk 1]
```diff
@@ -145,5 +147,24 @@ void foreach_tensor_lerp_ternary_xpu_(
   }
 }
 
+void foreach_tensor_copy_list_kernel_xpu_(
+    at::TensorList self,
+    at::TensorList src,
+    bool non_blocking) {
+      at::native::check_foreach_api_restrictions(self, src);
+      if (!at::native::can_use_fast_route(
```
Comment (majing921201) [inline src/ATen/native/xpu/ForeachOpList.cpp:155]: no need at::native, here is the same namespace

Comment (majing921201) [inline src/ATen/native/xpu/ForeachOpList.cpp:155]: The check condition seems not aligned CUDA, can you help check ?
https://github.com/pytorch/pytorch/blob/d8f22a114142f40f78e1145e45d7eaf823b516cb/aten/src/ATen/native/cuda/ForeachBinaryOpList.cu#L418

Comment (cfgfung) [reply src/ATen/native/xpu/ForeachOpList.cpp:155]: The extra checks by CUDA is caused by this feature enhancement - https://github.com/pytorch/pytorch/issues/115171
They wish to improve the GPU bandwidth utilization when src and dst are of different types.

Shall we stick to the current design as the copy kernel does not support that functionality?

Comment (cfgfung) [reply src/ATen/native/xpu/ForeachOpList.cpp:155]: Removed that from the code.

Comment (majing921201) [reply src/ATen/native/xpu/ForeachOpList.cpp]: > The extra checks by CUDA is caused by this feature enhancement - [pytorch/pytorch#115171](https://github.com/pytorch/pytorch/issues/115171) They wish to improve the GPU bandwidth utilization when src and dst are of different types.
> 
> Shall we stick to the current design as the copy kernel does not support that functionality?

if only for performance improvement, We could hold on the change.

**Review by majing921201 — APPROVED** [review]

**Pull request merged**